dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/trackedentity/DefaultTrackerAccessManager.java

```diff
@@ -107,6 +111,18 @@ public List<String> canWrite(UserDetails user, TrackedEntity trackedEntity) {
   @Override
   public List<String> canRead(
       UserDetails user, TrackedEntity trackedEntity, Program program, boolean skipOwnershipCheck) {
+    List<String> errors = canReadProgramAndTrackedEntityType(user, trackedEntity, program);
+
+    if (!skipOwnershipCheck && !ownershipAccessManager.hasAccess(user, trackedEntity, program)) {
```

Comment:
I'd say that considering this span for multiple programs with the same TET of the tracked entity, we can make the signature generic to accept a list of programs and iterate inside this method to get the list of errors. Or create another method with this behavior.

Reply:
or at least the _OWNERSHIP_ACCESS_DENIED_ error should indicate what is the program

Reply:
the thing with the error message is that in the end we just check if the returned list is empty or not, but we don't use the error messages within. That's also true for this case, where I just want to see if the user has access to at least one program, so the error message itself is not important.
I'd be in favor of start using the error messages, but that'd require a bit of a design, since we might break the API (specially when it comes to breaking the glass).